private/buf/bufwire/image_config.go

```diff
@@ -38,3 +38,10 @@ func (i *imageConfig) Image() bufimage.Image {
 func (i *imageConfig) Config() *bufconfig.Config {
 	return i.config
 }
+
+func NewImageConfig(image bufimage.Image, config *bufconfig.Config) *imageConfig {
```

Comment:
See the internal go style guide regarding file layout and where public types go, and see other places in this codebase for the patterns to follow here.